core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp

```diff
@@ -1227,13 +1260,23 @@ convert_r_i32_to_r_u16(x86::Assembler &a, int32 reg_no_dst, int32 reg_no_src)
  * @return  true if success, false otherwise
  */
 static bool
-convert_imm_i32_to_r_u64(x86::Assembler &a, int32 reg_no, int32 data)
+convert_imm_i32_to_r_i64(x86::Assembler &a, int32 reg_no, int32 data)
 {
-    return false;
+    int64 long_data = 0;
+    if (data & 0x80000000) {
+        long_data = data | 0xFFFFFFFF00000000;
+    }
+    else {
+        long_data = data & 0x00000000FFFFFFFF;
```

Comment:
Directly using `int64 long_data = (int64)data;` is OK, no need to convert it manually.

If not, should be 0x0xFFFFFFFF00000000LL and 0x00000000FFFFFFFFLL, or it might be treated as int32.

Reply:
`LL` is my mistake. 

not sure  `(int64)data` will let the compiler zero-extend or sign-extend the `data`, and make the code symmetrical, too

Reply:
In my memory, it depends on the type of data: (int64)(int32)data signed extends i32 to i64, (int64)(uint32)data unsigned extends i32 to i64.
If not very sure, using your original code is also OK, but should add LL suffix.

Reply:
a little research shows in the expression `int64 long_data = (int64)data`.

-if `data` is signed, *clang* will use `movsxd` to extend it. *gcc* will use `cdqe`.
-if `data` is unsigned, both will `mov` `data` to a 32bit reg, like *eax*, and then use the whole 64bit reg, like `rax`.

Reply:
OK, if the result isn't determinative, using the origin code (with LL) is better.